packages/core/src/render3/util/view_utils.ts

```diff
@@ -119,14 +121,14 @@ export function getNativeByTNodeOrNull(tNode: TNode|null, lView: LView): RNode|n
 
 export function getTNode(tView: TView, index: number): TNode {
   ngDevMode && assertGreaterThan(index, -1, 'wrong index for TNode');
-  ngDevMode && assertLessThan(index, tView.data.length - HEADER_OFFSET, 'wrong index for TNode');
-  return tView.data[index + HEADER_OFFSET] as TNode;
+  ngDevMode && assertLessThan(index, tView.data.length, 'wrong index for TNode');
```

Comment:
It looks like we can use the same checks like we have in `getNativeByIndex`?
```suggestion
  ngDevMode && assertIndexInRange(tView.data, index);
  ngDevMode && assertGreaterThanOrEqual(index, HEADER_OFFSET, 'Expected to be past HEADER_OFFSET');
```

Reply:
👍